### ISSUE TITLE
[BUG]: Guitar Hero Strummer & Whammer Bar

### ISSUE THREAD

**Issue by TheHobatron** [PCSX2/pcsx2#11011]
### Describe the Bug

I've connected an original PS2 guitar controller to my PC but having set the bindings, the strummer completely glitches out. It never stops moving on the menu screen, repeated attempts to reset it have not worked.

### Reproduction Steps

All I had to do was open the game and the menu glitch started immediately. Whammy bar just refuses to be set when I press it for calibration.

https://github.com/PCSX2/pcsx2/assets/165608827/2857c2ea-dcd4-41e1-9173-8223db6f752e

### Expected Behavior

I expected myself to be able to strum up and down as normal for fast songs, and use my whammy bar for extra points (and musicality!).

### PCSX2 Revision

v1.7.5661

### Operating System

Windows 11

### CPU

12th Gen Intel(R) Core(TM) i5-12600KF   3.70 GHz

### GPU

RTX 3070

**Comment by RedPanda4552** [comment]
TL;DR Using a real PS2 guitar with PCSX2 is likely never going to work until someone makes a proper adapter.

Currently all the "PS2 to USB" adapters you can get are no-name products which loosely implement the PS1 controller spec and do not attempt to implement all the commands which would identify a controller as a Dualshock 2. At least, they don't seem to, we can't really know because there's no documentation about any of these things, we can only go off of their outward behavior.

PS2 guitars require that whatever device they are talking to be able to comprehend a few differences in their configuration. They have no vibration motors, so when queried they will respond as such and the adapter needs to be able to understand this. They only have one axis, so when queried they will respond as such and the adapter needs to be able to understand this. If the adapter doesn't, it will either flat out not work or, more likely, just ignore whatever the guitar sent and keep acting like it's talking to a PS1 controller.

There is also the issue of the individual polls. A part of a guitar identifying itself to PS2 software is that the dpad left button is fused down. A normal game, if it doesn't realize it is talking to a guitar when all of its queries get nonstandard responses, will see the guitar as always sending dpad left. This extends to the adapter which is pretending to be a game running on a PS2 sending commands to the controller. If the adapter doesn't understand that it's talking to a guitar, it's going to always see dpad left held down, and start trying to apply that to every single one of your mappings no matter what.

The whammy bar can be confusing as well, because it is an axis which normally rests at the PS2 neutral position of 127 but then when actuated, moves towards 0. This one is a little less sensible and unclear, but we've generally noticed a pattern where some guitars still report their whammy bar as engaged even when it is resting at neutral. If I had to guess, this ends up just being a deadzone issue and some whammy bars are worn out, but is less clear.

Whichever way you cut it, someone needs to make an adapter which can send proper query commands to a guitar and properly report the pressed buttons, by understanding that it is talking to a guitar, not a PS1 controller. Otherwise putting a real PS2 guitar into your PC is never going to work right.
